### PR TITLE
[VL] Support regr_r2 aggregate function

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

Support regr_r2 aggregate function.

## How was this patch tested?

Add new test case in `VeloxAggregateFunctionsSuite`

### CODE REVIEW

**Comment by liujiayi771** [comment]
cc @rui-mo @PHILO-HE, thanks.

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala [hunk 1]
```diff
@@ -371,6 +371,26 @@ abstract class VeloxAggregateFunctionsSuite extends VeloxWholeStageTransformerSu
     }
   }
 
+  test("regr_r2") {
+    assume(isSparkVersionAtleast("3.3"))
```
Comment (PHILO-HE) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala:375]: Seems we can just use `testWithSpecifiedSparkVersion` to limit minimum Spark version?

**Review by PHILO-HE — COMMENTED** [review]
Just one comment. Thanks!

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala (hunk 1 above)
Comment (liujiayi771) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala:375]: `testWithSpecifiedSparkVersion` cannot express >= some version. Also, will there be any changes for Spark 3.3 and Spark 3.4 in the future? For example, upgrading from Spark 3.3.1 to 3.3.4.

Comment (PHILO-HE) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala:375]: @liujiayi771, cannot the below code work? It requires the minimum version=3.3, regardless of the third version digit. E.g., Spark's "3.3.1", "3.3.4" or "3.4.2" will pass the internal check (>= "3.3").
`testWithSpecifiedSparkVersion("name", "3.3")`

Comment (liujiayi771) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxAggregateFunctionsSuite.scala]: @PHILO-HE Yes, this can work, thanks.

**Review by rui-mo — APPROVED** [review]

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!
FYI. If the CH failed tests are the ones just fixed by https://github.com/apache/incubator-gluten/commit/131f70043004a299127d80ea080d07ff83231041, we can ignore the failure.

**Comment by liujiayi771** [comment]
> If the CH failed tests are the ones just fixed by [131f700](https://github.com/apache/incubator-gluten/commit/131f70043004a299127d80ea080d07ff83231041), we can ignore the failure.

Yes, we can ignore the failure.
```
[2024-04-01T09:30:12.651Z] - test mergetree table write *** FAILED ***
[2024-04-01T09:30:12.651Z]   org.apache.spark.SparkException: Job aborted.
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.errors.QueryExecutionErrors$.jobAbortedError(QueryExecutionErrors.scala:651)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.execution.datasources.v1.clickhouse.MergeTreeFileFormatWriter$.write(MergeTreeFileFormatWriter.scala:252)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.delta.ClickhouseOptimisticTransaction.$anonfun$writeFiles$1(ClickhouseOptimisticTransaction.scala:124)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.delta.ClickhouseOptimisticTransaction.writeFiles(ClickhouseOptimisticTransaction.scala:76)
[2024-04-01T09:30:12.651Z]   at org.apache.spark.sql.delta.files.TransactionalWrite.writeFiles(TransactionalWrite.scala:212)
```

**Pull request merged**